src/components/AccountSwitcher.tsx

```diff
@@ -0,0 +1,202 @@
+import React, {useRef, useState} from 'react';
+import {View} from 'react-native';
+import {useOnyx} from 'react-native-onyx';
+import useCurrentUserPersonalDetails from '@hooks/useCurrentUserPersonalDetails';
+import useLocalize from '@hooks/useLocalize';
+import useNetwork from '@hooks/useNetwork';
+import usePermissions from '@hooks/usePermissions';
+import useResponsiveLayout from '@hooks/useResponsiveLayout';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import {clearDelegatorErrors, connect, disconnect} from '@libs/actions/Delegate';
+import * as PersonalDetailsUtils from '@libs/PersonalDetailsUtils';
+import variables from '@styles/variables';
+import * as Modal from '@userActions/Modal';
+import CONST from '@src/CONST';
+import type {TranslationPaths} from '@src/languages/types';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {PersonalDetails} from '@src/types/onyx';
+import Avatar from './Avatar';
+import ConfirmModal from './ConfirmModal';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+import type {MenuItemProps} from './MenuItem';
+import MenuItemList from './MenuItemList';
+import type {MenuItemWithLink} from './MenuItemList';
+import Popover from './Popover';
+import {PressableWithFeedback} from './Pressable';
+import Text from './Text';
+
+function AccountSwitcher() {
+    const currentUserPersonalDetails = useCurrentUserPersonalDetails();
+    const styles = useThemeStyles();
+    const theme = useTheme();
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
+    const {canUseNewDotCopilot} = usePermissions();
+    const {shouldUseNarrowLayout} = useResponsiveLayout();
+    const [account] = useOnyx(ONYXKEYS.ACCOUNT);
+    const buttonRef = useRef<HTMLDivElement>(null);
+
+    const [shouldShowDelegatorMenu, setShouldShowDelegatorMenu] = useState(false);
+    const [shouldShowOfflineModal, setShouldShowOfflineModal] = useState(false);
+    const delegators = account?.delegatedAccess?.delegators ?? [];
+
+    const isActingAsDelegate = !!account?.delegatedAccess?.delegate ?? false;
+    const canSwitchAccounts = canUseNewDotCopilot && (delegators.length > 0 || isActingAsDelegate);
+
+    const createBaseMenuItem = (personalDetails: PersonalDetails | undefined, error?: TranslationPaths, additionalProps: MenuItemWithLink = {}): MenuItemWithLink => {
+        return {
+            title: personalDetails?.displayName ?? personalDetails?.login,
+            description: personalDetails?.login,
```

Comment:
Coming from https://github.com/Expensify/App/issues/48648, we should have used `Str.removeSMSDomain` to ensure phone numbers are displayed correctly (without the `@expensify.sms` suffix).